listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -293,6 +295,109 @@ def delete_feed_events(user_name):
         raise APIBadRequest(f"Invalid JSON: {str(e)}")
 
 
+@user_timeline_event_api_bp.route("/user/<user_name>/feed/events/hide", methods=['OPTIONS', 'POST'])
+@crossdomain
+@ratelimit()
+def hide_user_timeline_event(user_name):
+    '''
+    Hide events from the user feed, only recording_recommendation and recording_pin
+    events that have been generated by the people one is following can be deleted
+    via this endpoint
+    For example:
+
+    .. code-block:: json
+
+        {
+            "event_type": "recording_recommendation",
+            "event_id": "<integer id of the event>"
+        }
+
+    .. code-block:: json
+
+        {
+            "event_type": "recording_pin",
+            "event_id": "<integer id of the event>"
+        }
+
+    :param user_name: The MusicBrainz ID of the user from whose timeline events are being deleted
+    :type user_name: ``str``
+    :statuscode 200: Successful deletion
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized
+    :statuscode 404: User not found
+    :statuscode 500: API Internal Server Error
+    :resheader Content-Type: *application/json*
+    '''
+ 
+    user = validate_auth_header()
+    if user_name != user['musicbrainz_id']:
+        raise APIUnauthorized("You don't have permissions to hide events from this user's timeline.")
+
+    try:
+        data = ujson.loads(request.get_data())
+        if data["event_type"] == UserTimelineEventType.RECORDING_RECOMMENDATION.value:
+            row_id = data["event_id"]
+            result = db_user_timeline_event.get_user_timeline_events_by_id(row_id);
+            if db_user_relationship.is_following_user(user['id'], result.user_id):
+                if db_user_timeline_event.hide_user_timeline_event(user['id'], data["event_type"], data["event_id"]):
+                    return jsonify({"status": "ok"});
+                else:
+                    raise APIBadRequest("Event already hidden");
```

Comment:
@alastair thoughts on this? I'd say if user tries to hide twice, backend should ignore without raising APIBadRequest.